plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/editor/ERDEditorEmbedded.java

```diff
@@ -262,6 +270,29 @@ public void doSave(IProgressMonitor monitor) {
         }
         updateToolbarActions();
     }
+    
+    //Save local ERD diagram into a file
+    public void saveErdToFile(IProgressMonitor monitor) {
+        try {
+            String path = null;
+            FileDialog dialog = new FileDialog( parent.getShell(), SWT.SAVE );
+            String[] filterExt = {"*.erd"};
+            dialog.setFilterExtensions(filterExt);
+            path = dialog.open();
+            
+            if ( path != null ) {
+                String diagramState = DiagramLoader.serializeDiagram(RuntimeUtils.makeMonitor(monitor), getDiagramPart(), getDiagram(), false, true);
+
+                try (final OutputStreamWriter osw = new OutputStreamWriter(new FileOutputStream(path), GeneralUtils.UTF8_CHARSET)) {
+                    osw.write(diagramState);
+                }
+            }
+            else
+                log.error("Invalid path to save the image");
+        } catch (Exception e) {
+            log.error("Error saving diagram", e);
+        }
+    }
```

Comment:
What about moving this method into the action class itself? All required API is public, so no problems here.

Also, if there's no any way to obtain real monitor, use our `VoidProgressMonitor` instead to avoid unnecessary conversions.

Reply:
Understood, working on it

Reply:
Pardon the delay, yesterday got late, and I couldn't finish it, today ill put the last commits